src/results.cpp

```diff
@@ -590,17 +618,37 @@ Results Results::filter(Query&& q) const
     return Results(m_realm, get_query().and_query(std::move(q)), m_descriptor_ordering);
 }
 
-Results Results::distinct(realm::DistinctDescriptor&& uniqueness)
+Results Results::distinct(DistinctDescriptor&& uniqueness) const
 {
     DescriptorOrdering new_order = m_descriptor_ordering;
     new_order.append_distinct(std::move(uniqueness));
     return Results(m_realm, get_query(), std::move(new_order));
 }
 
+Results Results::distinct(std::vector<std::string> const& keypaths) const
+{
+    if (keypaths.empty())
+        return *this;
+    if (get_type() != PropertyType::Object) {
+        if (keypaths.size() != 1)
+            throw std::invalid_argument(util::format("Cannot sort array of '%1' on more than one key path",
```

Comment:
I kinda feel that using distinct as a verb is awkward enough that it's not worth making `parse_keypath()` use the correct word, and then concluded that these might as well be consistent with that. I don't feel strongly that this is the right thing to do, though.